src/hotspot/share/opto/castnode.cpp

```diff
@@ -126,37 +126,20 @@ uint ConstraintCastNode::size_of() const {
 Node* ConstraintCastNode::make_cast(int opcode, Node* c, Node* n, const Type* t, DependencyType dependency,
```

Comment:
I'm wondering if this method is still worth keeping as it would simply replace a "new CastXXNode" line which is just as expressive/readable. As far as I can see, it's currently only used with statically known Opcodes. So, we could replace them.

Reply:
On a separate note and might be worth to do in this change as well, we have `ConstraintCastNode::make_cast_for_type()` and `ConstraintCastNode::make()`. The intent of the former is clear but the latter switches on the provided basic type. Maybe we can rename the latter to `make_cast_for_basic_type()` to better align it with `make_cast_for_type()`.

Reply:
Agree that `make_cast` probably is not needed. And your renaming suggestion makes sense.

Can we have this in a separate PR? I generally prefer to keep PRs as small of a unit as possible.

Reply:
If you change the title of the PR into something like "cleanup CastNode construction" then I guess it's fine to do it all in once. But I leave it up to you to decide if you want to split it or not. Both is fine.